### PR TITLE
fix: don’t try to overwrite user id

### PR DESCRIPTION
**Pull request by pmuir**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

File: pkg/users/git.go [hunk 1]
```diff
@@ -71,6 +73,14 @@ func (r *GitUserResolver) Resolve(user *gits.GitUser) (*jenkinsv1.User, error) {
 		}
 		new := r.GitUserToUser(gitUser)
 		id = gitUser.Login
+		// Check if the user id is available, if not append "-<n>" where <n> is some integer
+		for i := 0; true; i++ {
+			_, err := r.JXClient.JenkinsV1().Users(r.Namespace).Get(id, v1.GetOptions{})
+			if err == nil {
```
Comment (davidcurrie) [inline pkg/users/git.go]: Shouldn't this be negated? We break out if we couldn't retrieve the user...

Better still is to use `errors.IsNotFound`:
https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L457

File: pkg/users/git_test.go [hunk 2]
```diff
@@ -202,6 +202,72 @@ func TestFindUserByFromGitProviderWithNoEmail(t *testing.T) {
...
+	}()
+	assert.NoError(t, err)
+	gitUser := gits.GitUser{
+		Login: gitUser1.Login,
+		Email: gitUser1.Email,
+	}
+	_, err = resolver.Resolve(&gitUser)
+	assert.NoError(t, err)
+	// Validate that we have both users
+	// TODO
```
Comment (davidcurrie) [inline pkg/users/git_test.go]: TODO ?!

Comment (pmuir) [reply pkg/users/git_test.go]: Ooops ;-)

File: pkg/users/git.go (hunk 1 above)
Comment (pmuir) [reply pkg/users/git.go]: Thanks, good point

**Pull request merged**